File: backend/src/main/java/com/tanerkaynar/nexuserp/repository/SiparisRepository.java
package com.tanerkaynar.nexuserp.repository;

import com.tanerkaynar.nexuserp.model.Siparis;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.List;

public interface SiparisRepository extends JpaRepository<Siparis, Integer> {
    List<Siparis> findTop50ByOrderBySiparisidDesc();
    List<Siparis> findByDurumOrderBySiparisidDesc(String durum);
}

File: client-mobile/NexusERP/app/src/main/java/com/tanerkaynar/nexuserp/data/repository/SiparisRepository.kt
package com.tanerkaynar.nexuserp.data.repository

import com.tanerkaynar.nexuserp.data.api.RetrofitClient
import com.tanerkaynar.nexuserp.data.model.Siparis
import com.tanerkaynar.nexuserp.data.model.SiparisRequest

class SiparisRepository {
    private val api get() = RetrofitClient.getApiService()

    suspend fun getAll(): Result<List<Siparis>> = safeCall { api.getSiparisler() }
    suspend fun getBekleyen(): Result<List<Siparis>> = safeCall { api.getBekleyenSiparisler() }
    suspend fun add(request: SiparisRequest): Result<Unit> = safeCall { api.addSiparis(request) }
    suspend fun sevkEt(id: Int): Result<Unit> = safeCall { api.sevkEt(id) }
    suspend fun update(id: Int, request: SiparisRequest): Result<Unit> = safeCall { api.updateSiparis(id, request) }
    suspend fun delete(id: Int): Result<Unit> = safeCall { api.deleteSiparis(id) }
}